src/node_types.cc

```diff
@@ -0,0 +1,69 @@
+#include "node_internals.h"
+
+using v8::Context;
+using v8::FunctionCallbackInfo;
+using v8::Local;
+using v8::Object;
+using v8::Value;
+
+namespace node {
+namespace {
+
+#define VALUE_METHOD_MAP(V)                                                   \
+  V(External)                                                                 \
+  V(Date)                                                                     \
+  V(ArgumentsObject)                                                          \
+  V(BooleanObject)                                                            \
+  V(NumberObject)                                                             \
+  V(StringObject)                                                             \
+  V(SymbolObject)                                                             \
+  V(NativeError)                                                              \
+  V(RegExp)                                                                   \
+  V(AsyncFunction)                                                            \
+  V(GeneratorFunction)                                                        \
+  V(GeneratorObject)                                                          \
+  V(Promise)                                                                  \
+  V(Map)                                                                      \
+  V(Set)                                                                      \
+  V(MapIterator)                                                              \
+  V(SetIterator)                                                              \
+  V(WeakMap)                                                                  \
+  V(WeakSet)                                                                  \
+  V(ArrayBuffer)                                                              \
+  V(DataView)                                                                 \
+  V(SharedArrayBuffer)                                                        \
+  V(Proxy)                                                                    \
+  V(WebAssemblyCompiledModule)                                                \
+
+
+#define V(type) \
+  static void Is##type(const FunctionCallbackInfo<Value>& args) {             \
+    args.GetReturnValue().Set(args[0]->Is##type());                           \
```

Comment:
If this gets exported and documented, there should be proper checks for the size of `args`, I suppose.

What would `util.types.isMap()` do without arguments?

Reply:
It would treat the argument as `undefined`. I’m not sure if this is the intended behavior.

Reply:
@TimothyGu Ah, that's good, then.